mlir/include/mlir-c/Dialect/Rock.h

```diff
@@ -0,0 +1,62 @@
+//===-- mlir-c/Dialect/Rock.h - C API for Rock dialect ---- C--===//
```

Comment:
Since this is actually going to be exported, I strongly suggest to rename.

Reply:
`Dialect.cpp` for the C API relating to [Dialect] is standard convention upstream

Reply:
But this is a very limited subset of Rock dialect, is `mlir-c` the best place to export it? I feel like this can be another orthogonal API like `Miir.h`. Eventually MIOpen is supposed to pull it in (one way or another) as well.

Reply:
Enabling c api for a dialect opens the opportunity to register the dialect. With this, user can use generic mlir c api functions to create a module using Rock dialect IR.

In addition to that, we're adding tuning API functions this time and I believe we'll keep adding external APIs via c-api in the future.

Reply:
Had another look into the miir.
I think one day we need to migrate the implementation of miir*() functions under lib/Dialect/Rock and expose them via mlirRock*() here.

Reply:
I think this is going to bring about some interesting discussion. Maybe we should have a meeting offline.